### ISSUE TITLE
Ability to disable the new "proc-macro is explicitly disabled" hint specifically

### ISSUE THREAD

**Issue by paul-hansen** [rust-lang/rust-analyzer#18414]
I just updated to the latest RA version and am now getting these hints:
```
macro-error: ::leptos::leptos_dom::tracing::instrument: proc-macro is explicitly disabled
macro-error: tracing::instrument: proc-macro is explicitly disabled
macro-error: server: proc-macro is explicitly disabled
```

Which is because I told it not to expand `instrument` and `server` macros. Since they just wrap the code, this allows RA to treat them as no-op and it helps with completion inside them, performance etc.

I'm fine with this hint existing, but I personally don't want it because it I use this in a lot of places and it creates a lot of noise. Unfortunately the only way I've found to disable it is to disable the `macro-error` diagnostic which seems overly broad and I'm afraid I'll miss other important macro errors.

There should probably be a way to disable that hint specifically. I'm open to other solutions too though, e.g. if there's a way to disable just hints for macro-error.

**Comment by futile** [comment]
Cross-posting [my comment in the PR](https://github.com/rust-lang/rust-analyzer/pull/18418#issuecomment-2577667217):

> Hi, just saw this PR, awesome work, thank you! Any idea why I get a diagnostic with "proc-macro is explicitly disabled" but kind macro-error, on version rust-analyzer 1.83.0 (90b35a62 2024-11-26)?

> If I add macro-error to my disabled diagnostics, it goes away. But adding proc-macro-disabled instead still shows the diagnostic.

> Any help would be appreciated, thanks! :)

**Comment by Veykril** [comment]
The kind seems to be `proc-macros-disabled` and `proc-macro-disabled`, there seems to be a mismatch between docs and reality

**Comment by Veykril** [comment]
Oh you are using the rustup component, no in that case the PR is not in there yet. The rustup version lags 6-12 weeks behind master

**Comment by futile** [comment]
Ohh got it, thanks a lot! Then it should be fixed with 1.84 I hope, got it! :)
